### PR TITLE
Fix stale deployment statuses

### LINKED ISSUE #13869: Confusing deployment status in the UI
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

I created a new deployment using `prefect deploy`. It has a worker polling it and runs that I kick off are picked up and complete.  However, the deployment's status is shown as Not Ready.  Given how prominently we show deployment status in the UI, this is pretty confusing. 

image

image

image

### Reproduction

```python3
So far I have seen this on OSS - I'll check against cloud and report back.
```

### Error

_No response_

### Versions

```Text
⇒  prefect version
Version:             3.0.0rc2
API version:         0.8.4
Python version:      3.12.3
Git commit:          4d70aefb
Built:               Thu, Jun 6, 2024 2:53 PM
OS/Arch:             darwin/x86_64
Profile:             orion
Server type:         ephemeral
Pydantic version:    2.7.3
Server:
  Database:          sqlite
  SQLite version:    3.45.3
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by collincchoy**
Similar (but different) to #14082 - this PR fixes a separate issue where Deployment statuses aren't being updated correctly. In particular, a work pool + queue with a running worker does not update a newly created deployment to READY. 

Closes #13869 

See https://github.com/PrefectHQ/prefect/pull/14097 for 2.x port.

### Checklist

- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
- [x] This pull request references any related issue by including "closes `<link to issue>`"
  
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/prefect/server/api/workers.py [hunk 1]
```diff
@@ -372,20 +372,19 @@ async def get_scheduled_flow_runs(
...
-        wq.id for wq in work_queues if wq.status == WorkQueueStatus.NOT_READY
-    ]
-
     background_tasks.add_task(
         mark_work_queues_ready,
-        polled_work_queue_ids=polled_work_queue_ids,
-        ready_work_queue_ids=ready_work_queue_ids,
+        polled_work_queue_ids=[
+            wq.id for wq in work_queues if wq.status != WorkQueueStatus.NOT_READY
+        ],
```
Comment (collincchoy) [inline src/prefect/server/api/workers.py:379]: this change is not strictly necessary but w/o it the DB has to run an update on last_polled for rows that match these IDs w/ status = NOT_READY _twice_. 
By splitting this list out, it reduces some redundancy.

also matches parity w/ Cloud.

File: tests/server/orchestration/api/test_workers.py [hunk 2]
```diff
@@ -1845,28 +1845,52 @@ async def test_updates_last_polled_on_a_multiple_work_queues(
                 assert work_queue.last_polled is None
 
     async def test_updates_last_polled_on_a_full_work_pool(
```
Comment (collincchoy) [inline tests/server/orchestration/api/test_workers.py:1847]: changes to this test adds coverage that all requested work queues' last polled field gets updated regardless of their status which covers my changes to the `polled_work_queue_ids` up in the app code.

File: tests/server/orchestration/api/test_workers.py [hunk 3]
```diff
@@ -1845,28 +1845,52 @@ async def test_updates_last_polled_on_a_multiple_work_queues(
...
         for work_queue in work_queues:
-            assert work_queue.last_polled is not None
+            assert (
+                work_queue.last_polled is not None
+            ), "Work queue should have updated last_polled"
             assert work_queue.last_polled > now
 
-    async def test_updates_work_queue_status_on_a_full_work_pool(
-        self, client, session, work_queues, work_pools
+    async def test_updates_statuses_on_a_full_work_pool(
```
Comment (collincchoy) [inline tests/server/orchestration/api/test_workers.py:1875]: this test now covers that both work queue statuses _and_ deployment statuses are updated here.

There's one more pre-existing test after this one that checks a single deployment is updated correctly but this covers that _multiple_ relevant deployments are updated regardless of work queue statuses

**Review by chrisguidry — APPROVED** [review]
Looks sharp to me!

File: src/prefect/server/api/workers.py [hunk 4]
```diff
@@ -372,20 +372,19 @@ async def get_scheduled_flow_runs(
...
+        ready_work_queue_ids=[
+            wq.id for wq in work_queues if wq.status == WorkQueueStatus.NOT_READY
+        ],
     )
 
     background_tasks.add_task(
         mark_deployments_ready,
-        work_queue_ids=ready_work_queue_ids,
+        work_queue_ids=[wq.id for wq in work_queues],
```
Comment (serinamarie) [inline src/prefect/server/api/workers.py:387]: nice

**Pull request merged**